op-monitorism/faultproof_withdrawals/dispute_game_helper.go

```diff
@@ -0,0 +1,355 @@
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+type DisputeGame struct {
+	disputeGameProxyAddress common.Address
+	//game object
+	faultDisputeGame *dispute.FaultDisputeGame
+
+	// game data
+	rootClaim     [32]byte
+	l2blockNumber *big.Int
+	l2ChainID     *big.Int
+}
+
+type DisputeGameHelper struct {
+	//objects
+	l1Client           *ethclient.Client
+	l2Client           *ethclient.Client
+	rpc_l2Client       *rpc.Client
+	optimismPortal2    *l1.OptimismPortal2
+	disputeGameFactory *dispute.DisputeGameFactory
+	ctx                context.Context
+	gameCache          *lru.Cache
+	rootProofCache     *lru.Cache
+}
+
+type OutputResponse struct {
+	Version    string `json:"version"`
+	OutputRoot string `json:"outputRoot"`
+}
+
+// Define the GameStatus type
+type GameStatus uint8
+
+// Define constants for the GameStatus using iota
+const (
+	IN_PROGRESS GameStatus = iota
+	CHALLENGER_WINS
+	DEFENDER_WINS
+)
+
+type DisputeGameFactoryCoordinates struct {
+	GameType                  uint32
+	GameIndex                 uint64
+	disputeGameProxyAddress   common.Address
+	disputeGameProxyTimestamp uint64
+}
+
+type DisputeFactoryGameHelper struct {
+	//objects
+	l1Client                 *ethclient.Client
+	optimismPortal2          *l1.OptimismPortal2
+	DisputeGameFactoryCaller dispute.DisputeGameFactoryCaller
+}
+
+type DisputeGameFactoryIterator struct {
+	DisputeGameFactoryCaller      *dispute.DisputeGameFactoryCaller
+	currentIndex                  uint64
+	gameCount                     uint64
+	init                          bool
+	DisputeGameFactoryCoordinates *DisputeGameFactoryCoordinates
+}
+
+// Implement the Stringer interface for pretty printing
+func (gs GameStatus) String() string {
+	switch gs {
+	case IN_PROGRESS:
+		return "IN_PROGRESS"
+	case CHALLENGER_WINS:
+		return "CHALLENGER_WINS"
+	case DEFENDER_WINS:
+		return "DEFENDER_WINS"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// Define the GameStatus type
+type GameCorrectness uint8
+
+// Define constants for the GameStatus using iota
+const (
+	UNKNOWN GameCorrectness = iota
+	CORRECT
+	INCORRECT
+)
+
+// Implement the Stringer interface for pretty printing
+func (gs GameCorrectness) String() string {
+	switch gs {
+	case UNKNOWN:
+		return "UNKNOWN"
+	case CORRECT:
+		return "CORRECT"
+	case INCORRECT:
+		return "INCORRECT"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+func (d DisputeGame) String() string {
+	return fmt.Sprintf("DisputeGame[ disputeGameProxyAddress=%v rootClaim=%s l2blockNumber=%s l2ChainID=%s ]",
+		d.disputeGameProxyAddress,
+		common.BytesToHash(d.rootClaim[:]).Hex(),
+		d.l2blockNumber.String(),
+		d.l2ChainID.String(),
+	)
+}
+
+func NewDisputeGameHelper(ctx context.Context, l1Client *ethclient.Client, l2Client *ethclient.Client, optimismPortal2 *l1.OptimismPortal2) (*DisputeGameHelper, error) {
+
+	gameCache, err := lru.New(1000)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cache: %w", err)
+	}
+	rootProofCache, err := lru.New(1000)
```

Comment:
The magic number 1000 should be defined as constant in the code in uppercase to improve the readibility of the code